src/util/pagination.js

```diff
@@ -51,9 +81,177 @@ async function offsetPaginate ({
+  let results = await queryBuilder.limit(nbResultsPerPage + 2)
+
+  const paginationMeta = getCursorPaginationMeta({
+    decodedCursor,
+    results,
+    startingAfter,
+    endingBefore,
+    nbResultsPerPage,
+  })
+
+  // remove cursor result
+  if (decodedCursor) results = results.filter(_.negate(matchCursorResult(decodedCursor)))
+
+  results = results.slice(0, nbResultsPerPage)
+  if (shouldReverseOrder) results = results.reverse()
+
+  const firstResult = _.first(results)
+  const lastResult = _.last(results)
+
+  paginationMeta.startCursor = firstResult ? createCursor(firstResult, cursorConfig) : null
+  paginationMeta.endCursor = lastResult ? createCursor(lastResult, cursorConfig) : null
+
+  paginationMeta.results = results
+  return paginationMeta
+}
+
+function getCursorPaginationMeta ({
+  decodedCursor,
+  results,
+  startingAfter,
+  endingBefore,
+  nbResultsPerPage,
+}) {
+  let hasPreviousPage = false
+  let hasNextPage = false
+
+  if (decodedCursor) {
+    const matchingResult = matchCursorResult(decodedCursor)
+
+    const cursorResult = results.find(matchingResult)
+    const resultsWithoutCursorResult = results.filter(_.negate(matchingResult))
+
+    if (startingAfter) {
+      hasPreviousPage = Boolean(cursorResult)
+      hasNextPage = resultsWithoutCursorResult.length > nbResultsPerPage
+    } else if (endingBefore) {
+      hasPreviousPage = resultsWithoutCursorResult.length > nbResultsPerPage
+      hasNextPage = Boolean(cursorResult)
+    }
+  } else {
+    if (endingBefore) hasPreviousPage = results.length > nbResultsPerPage
+    else hasNextPage = results.length > nbResultsPerPage
+  }
+
+  return {
+    hasPreviousPage,
+    hasNextPage,
+    nbResultsPerPage,
+  }
+}
+
+function applyCursorPaginationParameters ({
+  queryBuilder,
+  cursorConfig,
+  decodedCursor,
+  order,
+  shouldReverseOrder,
+}) {
+  return queryBuilder
+    .where(qb => {
+      const firstProp = cursorConfig[0].prop
+      const secondaryProp = cursorConfig.length === 2 ? cursorConfig[1].prop : null
+
+      let operator = order === 'asc' ? '>' : '<'
+      let operatorWithEqual = order === 'asc' ? '>=' : '<='
+
+      if (shouldReverseOrder) {
+        operator = reverseOperator(operator)
+        operatorWithEqual = reverseOperator(operatorWithEqual)
+      }
+
+      if (cursorConfig.length === 1) {
+        // if one-column cursor
+        // use operator with equal to include the cursor result
+        // to determine if there is a previous page
+        qb.where(firstProp, operatorWithEqual, decodedCursor[firstProp])
+      } else {
+        qb
+          .where(firstProp, operator, decodedCursor[firstProp])
+          .orWhere(qb2 => {
+            // if two-columns cursor
+            // use operator with equal on the secondary column to include the cursor result
+            // to determine if there is a previous page
+            return qb2
+              .where(firstProp, decodedCursor[firstProp])
```

Comment:
It’s not clear in `createCursor` function above that the strict `order` only applies to second property when two of them are provided, as a **tie breaker**.

This would explain why you only use `operatorWithEqual` with second property (and simple equality operator with first prop).

Reply:
Indeed, I'll improve that comment, thanks!